test/unit/host_discovered_test.rb

```diff
@@ -321,21 +321,6 @@ class HostDiscoveredTest < ActiveSupport::TestCase
     assert Token.where(:host_id => h.id).empty?
   end
 
-  test "all non-discovery facts are deleted after managed conversion" do
-    Setting[:discovery_clean_facts] = true
-    raw = parse_json_fixture('regular_host', true)
-    raw.merge!({
-      'delete_me' => "content",
-      'discovery_keep_me' => "content",
-      })
-    host = discover_host_from_facts(raw)
-    host.save
-    managed = ::ForemanDiscovery::HostConverter.to_managed(host)
-    managed.clear_facts
-    assert_nil managed.facts_hash['delete_me']
-    assert_equal "content", managed.facts_hash['discovery_keep_me']
-  end
-
```

Comment:
Please do not delete this test but rewrite it so the behavior is actually tested. What you want to do is to create some facts of a different type manually (create `FactValue` and `FactName` via FactoryBot) and test if these were actually removed.